pkg/controller/controller.go

```diff
@@ -863,11 +873,64 @@ func convertAndFilterCatalogToNamespacedTypes(namespace string, in *osb.CatalogR
 	return serviceClasses, servicePlans, nil
 }
 
+// generateEscapedName takes in an OSB ID and filters
+// it to fit K8S name restrictions. It escapes all
+// characters except for lowercase alphanumerics (excluding z
+// as we use that for the escape character), hyphens, and
+// periods, changing them into their UTF-8 hex
+// equivalents separated by 'z'. Hyphens that come after periods and
+// periods that come after hyphens or periods are also escaped.
+// If the resulting string is over the K8S limit of 63 characters,
+// it appends the md5 hash of the original OSB ID to the first 31 characters
+// of the escaped name, with trailing periods trimmed.
```

Comment:
not clear until reading later code that the trailing periods are that on the 31 chars, not the chars + md5.